Tests/MongoSwiftSyncTests/UnifiedTestRunner/SpecialTestOperations.swift

```diff
@@ -272,11 +273,10 @@ struct UnifiedTargetedFailPoint: UnifiedOperationProtocol {
         // The mongos on which to set the fail point is determined by the session argument (after resolution to a
         // session entity). Test runners MUST error if the session is not pinned to a mongos server at the time this
         // operation is executed.
-        expect(session.pinnedServerAddress)
-            .toNot(
-                beNil(),
-                description: "Session \(self.session) unexpectedly not pinned to a mongos. Path: \(context.path)"
-            )
+        guard session.pinnedServerAddress != nil else {
```

Comment:
switching to this approach as we otherwise crash on line 281 on Linux when the pinnedServerAddress is nil due to SWIFT-1146.
I ran into this when I was accidentally running a test that uses this assertion against serverless and it failed